### PR TITLE
[action] update_info_plist: Output error message if xcodeproj can not be found

### PR DESCRIPTION
**Pull request by obayer**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
Fixes: https://github.com/fastlane/fastlane/issues/13927

### Description

* Wrote a test to confirm defective behaviour
* Checked against project setup described in ticket and see if the `conversion of nil` error has been replaced by a more descriptive error message

### CODE REVIEW

**Comment by obayer** [comment]
I signed it!

File: fastlane/lib/fastlane/actions/update_info_plist.rb [hunk 1]
```diff
@@ -14,6 +14,7 @@ def self.run(params)
 
           # Assign folder from parameter or search for xcodeproj file
           folder = params[:xcodeproj] || Dir["*.xcodeproj"].first
+          UI.user_error!("You must specify an xcodeproj path") if folder.nil?
```
[the remark this answers is not in the document]
Comment (obayer) [reply fastlane/lib/fastlane/actions/update_info_plist.rb]: Feedback addressed @janpio

File: fastlane/lib/fastlane/actions/update_info_plist.rb [hunk 2]
```diff
@@ -14,6 +14,7 @@ def self.run(params)
 
           # Assign folder from parameter or search for xcodeproj file
           folder = params[:xcodeproj] || Dir["*.xcodeproj"].first
+          UI.user_error!("Could not figure out your xcodeproj path. Please specify it using the xcodeproj parameter") if folder.nil?
```
Comment (janpio) [inline fastlane/lib/fastlane/actions/update_info_plist.rb]:
```suggestion
           UI.user_error!("Could not figure out your xcodeproj path. Please specify it using the `xcodeproj` parameter") if folder.nil?
```

**Review by janpio — APPROVED** [review]
LGTM (with the last suggestion added)

**Comment by obayer** [comment]
@janpio is there anything I can or should do to get this PR merged?

**Pull request merged**